src/algorithms/cfd/fd_first_algorithm.cpp

```diff
@@ -0,0 +1,586 @@
+            continue;
+        const Itemset newset = Join(iset, jnode.item);
+        auto c = ConstructIntersection(inode.candidates, jnode.candidates);
+        for (int zz : newset) {
+            const auto zsub = ConstructSubset(newset, zz);
+            const auto stored_sub = store_.find(zsub);
+            if (stored_sub == store_.end()) {
+                c.clear();
+                break;
+            }
+            const Itemset& sub_cands = *cand_store_.Find(zsub);
+            c = ConstructIntersection(c, sub_cands);
+        }
+        if (!c.empty()) {
+            expands.emplace_back(&items[jx].tids);
+            auto new_node = MinerNode<PartitionTIdList>(items[jx].item);
+            new_node.candidates = c;
+            new_node.prefix = ConstructSubset(newset, new_node.item);
+            tmp_suffix.push_back(std::move(new_node));
+        }
+    }
+    return {expands, tmp_suffix};
+}
+
+void FDFirstAlgorithm::FdsFirstDFS(const Itemset& prefix, const PIdListMiners& items,
+                                   Substrategy ss) {
+    for (int ix = static_cast<int>(items.size()) - 1; ix >= 0; ix--) {
+        const MinerNode<PartitionTIdList>& inode = items[ix];
+        const Itemset iset = Join(prefix, inode.item);
+        const auto insect = ConstructIntersection(iset, inode.candidates);
+        for (int out : insect) {
+            const Itemset sub = ConstructSubset(iset, out);
+            MineFD(inode, sub, out);
+
+            if (ss == +Substrategy::dfs) {
+                MinePatternsDFS(sub, out, inode.tids);
+            } else if (ss == +Substrategy::bfs) {
+                MinePatternsBFS(sub, out, inode.tids);
+            }
+        }
+
+        if (inode.candidates.empty()) continue;
+        if (iset.size() == max_cfd_size_) continue;
+        store_[iset] = inode.tids;
+        cand_store_.Insert(iset, inode.candidates);
+
+        const auto [expands, tmp_suffix] = ExpandMiningFd(inode, ix, iset, items);
+
+        const auto exps = PartitionTIdListUtil::ConstructIntersection(items[ix].tids, expands);
+        PIdListMiners suffix;
+        for (size_t e = 0; e < exps.size(); e++) {
+            bool gen = true;
+            const auto new_set = Join(tmp_suffix[e].prefix, tmp_suffix[e].item);
+            const auto sp = std::make_pair(TIdUtil::Support(exps[e]), exps[e].sets_number);
+            const auto free_map_elem = free_map_.find(sp);
+            if (free_map_elem != free_map_.end()) {
+                const auto free_cands = free_map_elem->second;
+                for (const auto& sub_cand : free_cands) {
+                    if (IsSubsetOf(sub_cand, new_set)) {
+                        gen = false;
+                        break;
+                    }
+                }
+            }
+            if (gen) {
+                free_map_[sp].push_back(new_set);
+                free_itemsets_.insert(new_set);
+            }
+            // MinerNode(Item, PartitionTIdList)
```

Comment:
The only thing left is to get rid of such comments, i guess they were used during coding, and now have no purpose.